pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig_test.go

```diff
@@ -0,0 +1,204 @@
+package bootstrapperconfig
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/scheme/fake"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	dtwebhook "github.com/Dynatrace/dynatrace-operator/pkg/webhook"
+	dtclientmock "github.com/Dynatrace/dynatrace-operator/test/mocks/pkg/clients/dynatrace"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	testPaasToken       = "test-paas-token"
+	testAPIToken        = "test-api-token"
+	testDataIngestToken = "test-ingest-token"
+
+	testUUID                  = "test-uuid"
+	testTenantToken           = "abcd"
+	testCommunicationEndpoint = "https://tenant.dev.dynatracelabs.com:443"
+
+	testHost = "test-host"
+
+	testDynakube   = "test-dynakube"
+	testNamespace  = "test-namespace"
+	testNamespace2 = "test-namespace2"
+
+	testNamespaceDynatrace = "dynatrace"
+
+	testApiUrl = "https://" + testHost + "/e/" + testUUID + "/api"
+)
+
+func TestNewSecretGenerator(t *testing.T) {
+	client := fake.NewClient()
+	mockDTClient := dtclientmock.NewClient(t)
+
+	secretGenerator := NewSecretGenerator(client, client, mockDTClient)
+	assert.NotNil(t, secretGenerator)
+
+	assert.Equal(t, client, secretGenerator.client)
+	assert.Equal(t, client, secretGenerator.apiReader)
+	assert.Equal(t, mockDTClient, secretGenerator.dtClient)
+}
+
+func TestGenerateForDynakube(t *testing.T) {
+	t.Run("succcessfully generate secret for dynakube", func(t *testing.T) {
+		dynakube := &dynakube.DynaKube{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testDynakube,
+				Namespace: testNamespaceDynatrace,
+			},
+			Spec: dynakube.DynaKubeSpec{
+				APIURL: testApiUrl,
+			},
+		}
+
+		clt := fake.NewClientWithIndex(
+			dynakube,
+			clientInjectedNamespace(testNamespace, testDynakube),
+			clientSecret(testDynakube, testNamespaceDynatrace, map[string][]byte{
+				dtclient.ApiToken:  []byte(testAPIToken),
+				dtclient.PaasToken: []byte(testPaasToken),
+			}),
+			clientSecret(dynakube.OneAgent().GetTenantSecret(), testNamespaceDynatrace, map[string][]byte{
+				"tenant-token": []byte(testTenantToken),
+			}),
+			clientSecret(consts.BootstrapperInitSecretName, testNamespace, nil),
+		)
+
+		mockDTClient := dtclientmock.NewClient(t)
+
+		mockDTClient.On("GetProcessModuleConfig", mock.AnythingOfType("context.backgroundCtx"), mock.AnythingOfType("uint")).Return(&dtclient.ProcessModuleConfig{}, nil)
+
+		secretGenerator := NewSecretGenerator(clt, clt, mockDTClient)
+		err := secretGenerator.GenerateForDynakube(context.Background(), dynakube)
+		require.NoError(t, err)
```

Comment:
why not verify what is in the secret ? or the fact that it exists?
- you setup the `clt` so it is not much more work

the following code would also pass this test:
```go
func (g SecretGenerator) GenerateForDynakube(_ context.Context, _ dynakube.DynaKube) error {
	return nil
}
```

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/caf3dcbdf47a53cbf539a588a930e1dc1c10eee8